x-pack/solutions/security/plugins/security_solution/common/endpoint/index_data.ts

```diff
@@ -135,14 +135,16 @@ export const indexHostsAndAlerts = usageTracker.track(
 
       mergeAndAppendArrays(response, indexedHosts);
 
-      await indexAlerts({
-        client,
-        eventIndex,
-        alertIndex,
-        generator,
-        numAlerts: alertsPerHost,
-        options,
-      });
+      if (alertsPerHost > 0) {
```

Comment:
Q: I'm curious, why is this check, `alertsPerHost >0`, needed?

Reply:
`indexAlerts()` seem to have been called even if `alertsPerHost` was `0`, so I just place a conditional around it so we don't actually call the function when we don't want alerts to be generated